lib/actuator/LiftsLimitSwitchesManager.cpp

```diff
@@ -0,0 +1,99 @@
+#include "actuator/LiftsLimitSwitchesManager.hpp"
+#include <thread.h>
+#include <iostream>
+
+namespace cogip {
+namespace actuators {
+namespace positional_actuators {
+
+LiftsLimitSwitchesManager& LiftsLimitSwitchesManager::instance() {
+    static LiftsLimitSwitchesManager inst;
+    return inst;
+}
+
+void LiftsLimitSwitchesManager::init() {
+    // Create the thread to handle GPIO events
+    event_thread_pid_ = thread_create(
+        event_stack_, sizeof(event_stack_),
+        THREAD_PRIORITY_MAIN - 1,
+        THREAD_CREATE_STACKTEST,
+        event_thread_entry, nullptr, "gpio_evt"
+    );
+
+    std::cout << "Motor limit switches manager started" << std::endl;
+}
+
+void LiftsLimitSwitchesManager::register_gpio(gpio_t pin, cogip::actuators::positional_actuators::Lift* lift) {
```

Comment:
You can return appropriates error codes here, it can be useful for error management.

Reply:
Done